curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-intermediate-oop-by-building-a-platformer-game/64c9db021d4d912906878f3a.md

```diff
@@ -0,0 +1,334 @@
+---
+id: 64c9db021d4d912906878f3a
+title: Step 75
+challengeType: 0
+dashedName: step-75
+---
+
+# --description--
+
+Inside your condition, add a `forEach` loop to iterate through the `platforms` array. 
+
+Inside the loop, use the subtraction assignment operator to subtract 5 from the platform's `x` position.
+
+# --hints--
+
+You should have a `forEach` loop that iterates through the `platforms` array.
+
+```js
+assert.match(code, /platforms\.forEach\(\s*(.*?)\s*=>\s*{\s*(.*?)\s*}\s*\);?/);
```

Comment:
The test is always satisfied because it matches the previous forEach. This should work:
```suggestion
assert.match(code, /(?<=if\s*\(.*\)\s*{\s+)platforms\.forEach\(\s*\(.*\)\s*=>\s*{\s*(.*?)\s*}\s*\);?/);
```